com.unity.mobile.notifications/Runtime/Android/Plugins/com/unity/androidnotifications/UnityNotificationManager.java

```diff
@@ -539,7 +542,10 @@ protected static void sendNotification(Context context, Intent intent) {
 
     // Call the system notification service to notify the notification.
     protected static void notify(Context context, int id, Notification notification) {
-        getNotificationManager(context).notify(id, notification);
+        boolean showInForeground = notification.extras.getBoolean("showInForeground", false);
+        if (!isInForeground() || showInForeground) {
```

Comment:
This isn't entirely consistent with iOS behavior. On iOS notification is not shown, but the callback is still called informing about the arrival of notification.